src/components/Preview/Modal.tsx

```diff
@@ -117,11 +117,16 @@ export const Modal: FC<Props> = ({ open, onCancel, onOk }) => {
     setImage(svgElementToString(el));
   }, [background, image]);
 
-  const onDownload = async (format: ExportType) => {
+  const onDownload = async (exportType: ExportType) => {
     const el = stringToSVGElement(image as string);
     const bg = el.querySelector(".ogma-svg-background") as SVGRectElement;
     bg!.setAttribute("fill-opacity", background ? "1" : "0");
-    await handleDownload(el, format, visualisation.title);
+    // We send a telemetry event if the access to this app is as an LKE module and not as a plugin
+    const accessFromModules = window.location.pathname.includes("modules");
```

Comment:
can you explain to me how this works? Also, this seems a bit fragile, is there no other way? What if we change the string 'modules' by 'random', will this fail silently?

Reply:
To access the plugin now we have two ways:

1. Either as a plugin, as we access it via custom actions, in this way we access the plugin using "plugins/image-export-plugin"
2. Or, the new way for the export in LKE, we have a built version of the plugin in the frontend repositories under "modules/module-image-export", this way will be the same in the future for any embedded plugin into LKE, where we access in the path modules/lke-plugin-etc, for reference check this PR: https://github.com/Linkurious/linkurious-frontend/pull/5592 , if you have any question let me know

So in this case we can access the plugin either having "plugins" in the path or "modules" and these two ways are static and we rely on them in several places. We need to send the event if we are using the plugin as an embedded plugin in LKE and not as a "Plugin", checking the path should be enough, in the level of the  plugin we don't have any info how we get into it unless the path.
If we change the modules to random, the accessFromModules will be false and we don't send an event.

Reply:
but you see my issue? What if someday, in the frontend we change the code to change the url from 'modules' to 'randomString'. In that case, we would stop sending the event and will probably never notice because it's a different project. Is there a better way than checking the url?

Reply:
I see, but we are strongly dependent to the "modules" folder path.
Not only in the frontend but even in the server in this PR: https://github.com/Linkurious/linkurious-server/pull/4832/files
The folder modules is not created by the dev, it is automatically created in buildEnv where we copy the plugin into a folder called modules, and we added the "modules" in order to have some fix for the basePath (see screenshot, we have two level of parents which is static here too).
If "modules" by mistake is changed , the plugin itself will not be opened in order to send the events, as statically we use "modules" in frontend and backend, I don't see an issue using it here.
![Screen Shot 2024-08-09 at 11 43 19 AM](https://github.com/user-attachments/assets/223297df-17d5-4043-bb75-2db82a67d305)

Moreover, we don't have a way in the plugin to know but the URl either by checking the path OR if we need to send an extra param as a flag to send/not send tm events, but we still relying on this.

Relyin on "modules" is similar to relying on "plugins" for the path.

Reply:
Approving since there are no better way. However it's exactly the same as the 'plugin'. If the 'plugin' is changed, the plugin won't work anymore and we will notice. The issue for this, is that it fails silently, but not a big deal and I agree it has a pretty low risk to happen.